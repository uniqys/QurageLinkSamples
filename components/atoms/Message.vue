<template>
  <div v-if="data">
    {{ data.sender }}: {{ data.contents }}<br>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator'
import { Message as MessageData } from '../../types'

@Component
export default class Message extends Vue {
  @Prop() data!: MessageData | null
}
</script>
