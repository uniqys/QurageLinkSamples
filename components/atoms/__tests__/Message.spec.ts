import { shallowMount } from '@vue/test-utils'
import Message from '~/components/atoms/Message.vue'

describe('Message.vue', () => {
  it('should to match snapshot', () => {
    const wrapper = shallowMount(Message, {
      propsData: {
        data: { sender: '0x5c0bA9fC40f672932FaC478ca8B69d855C922675', contents: 'Hello, World!' }
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
