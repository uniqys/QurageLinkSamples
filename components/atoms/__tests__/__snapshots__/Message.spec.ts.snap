// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message.vue should to match snapshot 1`] = `
<div>
  
  0x5c0bA9fC40f672932FaC478ca8B69d855C922675: Hello, World!
  <br />
</div>
`;
